src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -92,7 +92,6 @@ public static function getSubscribedServices()
             'request_stack' => '?'.RequestStack::class,
             'http_kernel' => '?'.HttpKernelInterface::class,
             'serializer' => '?'.SerializerInterface::class,
-            'session' => '?'.SessionInterface::class,
```

Comment:
Is it a BC break? container is protected

Reply:
a behavioral one,  yes :) as long as `session` is only deprecated it should remain accessible here.

Reply:
Needs help to fix this. Issue is:
- `session` and `SessionInterface` are deprecated
- `AbstractController` exposes a container with session inside (whether or not the user use it.)

When we deprecate the service `session`, people still have a deprecation notice because the all the controllers subscribes to it.

Should we provide a "DeprecationContainer" that trigger deprecation when people call "get" at runtime?

Reply:
`AbstractController` exposes a service locator with _a_ `session` service. This does not necessarily have to be _the_ `session` service. You could create a new service with a factory that fetches the session from the request stack and additionally triggers a deprecation. This new services is then linked into the locator as `session` service. This way, you should only get a deprecation is the session service is actually accessed.